File: eidocell-backend/schemas/learning.py
from datetime import datetime
from pydantic import BaseModel, Field


class TrainModelRequest(BaseModel):
    name: str
    classifier_type: str = "knn"
    feature_source: str = "raw"
    params: dict = {}


class TrainModelResult(BaseModel):
    model_id: str
    name: str
    classifier_type: str
    feature_source: str
    n_classes: int
    n_training_samples: int
    accuracy: float | None
    label_mapping: dict


class RunInferenceRequest(BaseModel):
    model_id: str


class InferenceResult(BaseModel):
    total_predicted: int
    classes_created: list[str]
    assignments: dict[str, int]


class ApplyModelRequest(BaseModel):
    target_session_id: str


class TrainedModelOut(BaseModel):
    id: str
    name: str
    classifier_type: str
    feature_source: str
    feature_dim: int
    n_classes: int
    n_training_samples: int
    accuracy: float | None
    label_mapping: dict
    session_id: str
    created_at: datetime

    model_config = {"from_attributes": True}

File: eidocell-ui/src/types/learning.ts
export interface TrainModelRequest {
  name: string
  classifier_type?: string
  feature_source?: string
  params?: Record<string, unknown>
}

export interface TrainModelResult {
  model_id: string
  name: string
  classifier_type: string
  feature_source: string
  n_classes: number
  n_training_samples: number
  accuracy: number | null
  label_mapping: Record<string, string>
}

export interface RunInferenceRequest {
  model_id: string
}

export interface InferenceResult {
  total_predicted: number
  classes_created: string[]
  assignments: Record<string, number>
}

export interface ApplyModelRequest {
  target_session_id: string
}

export interface TrainedModelOut {
  id: string
  name: string
  classifier_type: string
  feature_source: string
  feature_dim: number
  n_classes: number
  n_training_samples: number
  accuracy: number | null
  label_mapping: Record<string, string>
  session_id: string
  created_at: string
}
